contrib_bots/lib/spotify.py

```diff
@@ -0,0 +1,159 @@
+# -*- coding: utf-8 -*-
+from future import standard_library # You would need to download future.
+standard_library.install_aliases()
+import json, urllib.request, urllib.parse, urllib.error
+BASE_URL = 'https://api.spotify.com/v1/search?q='
+class SpotifyHandler(object):
+    '''
+    This bot will allow users to directly search
+    for songs, artists, and albums using the Spotify API.
+    It can be called using `@spotify song/artist/album <your_search_query>`
+    '''
+    def usage(self):
+        return '''
+        This bot will allow users to directly search
+        for songs, artists, and albums using the Spotify API.
+        It can be called using `@spotify song/artist/album <your_search_query>`
+        This will return the top 10 search results from Spotify for either
+        song/artist/album.
+            '''
+
+    def triage_message(self, message):
+        # return True iff we want to (possibly) response to this message
+        original_content = message['content']
+        # This next line of code is defensive, as we
+        # never want to get into an infinite loop of posting follow
+        # ups for own follow ups!
+        if message['display_recipient'] == 'spotify':
+            return False
+        is_spotify = original_content.startswith('@spotify')
+
+        return is_spotify
+
+    def handle_message(self, message, client, state_handler):
+
+        def get_data(query_type, original_content):
+            if query_type == "song":
+                content_limit = 5
+                query_type = "track"
+            elif query_type == "album":
+                content_limit = 6
+            elif query_type == "artist":
+                content_limit = 7
+
+            encoded_message = urllib.parse.quote(original_content[content_limit:])
+            url = "{}{}&type={}&limit=10".format(BASE_URL, encoded_message, query_type)
+            return urllib.request.urlopen(url).read()
+        def get_data_multiple_parameters(query_data):
+            url = BASE_URL
+            for paramter in query_data:
+                url+="{}:{}%20".format(paramter[0],paramter[1])
```

Comment:
That space is to separate the different parameters. You can see it in this (example)[https://developer.spotify.com/web-api/console/get-search-item/?q=album%3Aarrival+artist%3Aabba&type=album].